akka-http-core/src/test/scala/akka/http/impl/engine/client/TlsEndpointVerificationSpec.scala

```diff
@@ -47,6 +51,12 @@ class TlsEndpointVerificationSpec extends AkkaSpec("""
         e shouldBe an[Exception]
```

Comment:
Right, that addresses my other question actually :) So when no host given, we're the server and skip the validation. Now the test passes.